examples/webapp/webapp.py

```diff
@@ -0,0 +1,202 @@
+
+STARTING_TIMEOUT = timedelta(minutes=4)
+
+
+class HttpService(HttpProxyService):
+    def __init__(self, db_address: str, db_port: int = 4001):
+        super().__init__(remote_port=5000)
+        self._db_address = db_address
+        self._db_port = db_port
+
+    @staticmethod
+    async def get_payload():
+        return await vm.repo(
+            image_hash=HTTP_IMAGE_HASH,
+            capabilities=[vm.VM_CAPS_VPN],
+        )
+
+    async def start(self):
+        # perform the initialization of the Service
+        # (which includes sending the network details within the `deploy` command)
+        async for script in super().start():
+            yield script
+
+        script = self._ctx.new_script(timeout=timedelta(seconds=10))
+
+        script.run(
+            "/bin/bash",
+            "-c",
+            f"cd /webapp && python app.py --db-address {self._db_address} --db-port {self._db_port} initdb",
+        )
+        script.run(
+            "/bin/bash",
+            "-c",
+            f"cd /webapp && python app.py --db-address {self._db_address} --db-port {self._db_port} run > /webapp/out 2> /webapp/err &",
+        )
+        yield script
+
+    async def reset(self):
+        # We don't have to do anything when the service is restarted
+        pass
+
+
+class DbService(Service):
+    def __init__(self):
+        super().__init__()
+
+    @staticmethod
+    async def get_payload():
+        return await vm.repo(
+            image_hash=DB_IMAGE_HASH,
+            capabilities=[vm.VM_CAPS_VPN],
+        )
+
+    async def start(self):
+        # perform the initialization of the Service
+        # (which includes sending the network details within the `deploy` command)
+        async for script in super().start():
+            yield script
+
+        script = self._ctx.new_script(timeout=timedelta(seconds=30))
+        script.run("/bin/run_rqlite.sh")
+        yield script
+
+    async def reset(self):
+        # We don't have to do anything when the service is restarted
+        pass
+
+
+async def main(subnet_tag, payment_driver, payment_network, port):
+    async with Golem(
+        budget=1.0,
+        subnet_tag=subnet_tag,
+        payment_driver=payment_driver,
+        payment_network=payment_network,
+    ) as golem:
+        print_env_info(golem)
+
+        network = await golem.create_network("192.168.0.1/24")
+        async with network:
+            db_cluster = await golem.run_service(DbService, network=network)
+            db_instance = db_cluster.instances[0]
+
+            while db_instance.state != ServiceState.running:
+                await asyncio.sleep(5)
+                print(db_instance)
+
+            print(
+                f"{TEXT_COLOR_CYAN}DB instance started, spawning the web server{TEXT_COLOR_DEFAULT}"
+            )
+
+            commissioning_time = datetime.now()
+
+            web_cluster = await golem.run_service(
```

Comment:
Maybe we could start both clusters at the same time? We'd only need to assign some IP to the DB instance, and pass it here again, I guess?
Starting both clusters at the same time would be super useful & matches the intuition & also matches what we'd like to have in the future `golem-compose` runner.

Reply:
hmm, I'm unsure if it's possible right now so guarantee that in case of failure, the address will be what we want... which would, of course, mean that we have an issue to solve but I wouldn't like to solve that before we have an otherwise working example.... 

but yes, I agree that's the desired scenario

Reply:
I don't understand what is the issue.
Is it that if the DB fails to start we'll get another one with a different IP?

Reply:
by default, that's how it works

Reply:
Are you sure?

I looked briefly in the code and I'd say that if `ServiceRunner.add_instance` is called with a `network_address`, it will either spawn an instance with this address, or will raise a `NetworkError`. And `ServiceRunner.add_instance` will not be called more than once.

But maybe I'm wrong, i didn't check this carefully.

Reply:
I'm not sure, just pretty sure... but anyway, as I said, I'd rather not assume it before we verify it